whelktool/src/main/groovy/whelk/datatool/form/MatchForm.groovy

```diff
@@ -0,0 +1,307 @@
+package whelk.datatool.form
+
+import groovy.transform.Memoized
+import whelk.Document
+import whelk.JsonLd
+import whelk.Whelk
+import whelk.datatool.util.DocumentComparator
+import whelk.datatool.util.IdLoader
+import whelk.util.DocumentUtil
+
+import static whelk.JsonLd.GRAPH_KEY
+import static whelk.JsonLd.ID_KEY
+import static whelk.JsonLd.RECORD_KEY
+import static whelk.JsonLd.RECORD_TYPE
+import static whelk.JsonLd.THING_KEY
+import static whelk.JsonLd.TYPE_KEY
+import static whelk.JsonLd.asList
+import static whelk.component.SparqlQueryClient.GRAPH_VAR
+import static whelk.converter.JsonLDTurtleConverter.toTurtleData
+import static whelk.util.DocumentUtil.getAtPath
+import static whelk.util.LegacyIntegrationTools.getMarcCollectionInHierarchy
+
+class MatchForm {
+    private static final DocumentComparator comparator = new DocumentComparator()
+
+    public static final String MATCHING_MODE = 'bulk:matchingMode'
+    public static final String HAS_ID = 'bulk:hasId'
+    public static final String BNODE_ID = 'bulk:formBlankNodeId'
+    public static final String ANY_TYPE = "bulk:Any"
+    public static final String SUBTYPES = "bulk:Subtypes"
+    public static final String EXACT = 'bulk:Exact'
+
+    private static final String VALUE = 'value'
+    private static final String VALUE_FROM = 'bulk:valueFrom'
+    private static final String ANY_OF = 'bulk:AnyOf'
+    private static final String HAS_BASE_TYPE_TMP = '_hasBaseTypeTmp'
+
+    Map form
+
+    Map<String, List> nodeIdToMatchFormPath
+    Map<String, Set<String>> nodeIdMappings
+    Map<String, Set<String>> baseTypeMappings
```

Comment:
I think all these need a short description.
What do they contain?

(In the long run we should try to move away from using strings everywhere. With proper types these could maybe be self explanatory)